digital-deck-ui/src/store/index.js

```diff
@@ -0,0 +1,77 @@
+import { createStore } from 'vuex';
+import axios from 'axios';
+import UnitTests from './../scripts/UnitTests.js';
+
+export default createStore({
+    state: {
+        sessionId: -1,
+        playerId: -1,
+        playerCards: null,
+        tableCards: null,
+        numCardsInDeck: -1
+    },
+    mutations: {
+        setSessionId(state, sessionId) {
+            state.sessionId = sessionId;
+        },
+
+        setPlayerId(state, playerId) {
+            state.playerId = playerId;
+        },
+
+        setPlayerCards(state, playerCards) {
+            state.playerCards = playerCards;
+        },
+
+        setTableCards(state, table) {
+            state.tableCards = table.cards;
+        },
+
+        setCardsInDeck(state, deck) {
+            state.numCardsInDeck = deck.length;
+        }
+    },
+    actions: {
+        initSession({ commit, state }) {
+            return axios.post('http://localhost:5000/new-session', {
+                decks: 1,
+                players: 1,
+                cardsPerPlayer: 1,
+                cardsOnTable: 0
+            }).then((res) => {
+                commit('setSessionId', res.data.sessionId);
+                commit('setPlayerId', res.data.players[0].playerId);
+                commit('setPlayerCards', res.data.players[0].cards);
+                commit('setTableCards', res.data.table);
+                commit('setCardsInDeck', res.data.deck);
+
+                UnitTests.testInitSession(state);
```

Comment:
assuming these will be removed or are not considered when building as non-dev?

Reply:
Yeah. We could have a flag to set to ignore all tests when merging into main.